File: server/models/address.js
const mongoose = require("mongoose");

const addressSchema = mongoose.Schema(
  {
    phoneNumber: {
      type: Number,
      required: [true, "Address must have a phone number!"],
    },
    detailedAddress: {
      type: String,
      required: [true, "Address must have a detailed address!"],
    },
    district: {
      type: String,
      required: [true, "Address must have district!"],
    },
    city: {
      type: String,
      required: [true, "Address must have city!"],
    },
    country: {
      type: String,
      required: [true, "Address must have country!"],
    },
    isDefault: {
      type: Boolean,
      default: false,
    },
  },
  {
    timestamps: true,
  }
);

const Address = mongoose.model("Address", addressSchema);
module.exports = Address;

File: lib/features/auth/domain/entities/address.dart
import 'package:equatable/equatable.dart';

class Address extends Equatable {
  final int phoneNumber;
  final String district;
  final String city;
  final String country;
  final String detailAddress;
  final bool isDefault;

  const Address({
    required this.phoneNumber,
    required this.district,
    required this.city,
    required this.country,
    required this.detailAddress,
    required this.isDefault,
  });

  @override
  List<Object?> get props => [
        phoneNumber,
        district,
        city,
        country,
      ];
}
